setup_chroot.sh

```diff
@@ -83,12 +83,13 @@ build_chroot()
 	echo -e "\n${COLOR_ON}Creating /etc/schroot/chroot.d/${CHROOT_NAME}.conf...${COLOR_OFF}" 
 	printf "[${CHROOT_NAME}]\ndescription=Ubuntu 12.04 Precise for ${pkg}\ndirectory=${CHROOT_DIR}/${CHROOT_NAME}\npersonality=${personality}\ngroups=sudo\nroot-groups=sudo\npreserve-environment=true\ntype=directory\n" | sudo tee /etc/schroot/chroot.d/${CHROOT_NAME}.conf
 
-	# Add the Ubuntu GPG key to apt
-	sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 0x40976EAF437D05B5
+	# Get the Ubuntu GPG key for Precise packages
+	PRECISE_KEYRING=/tmp/ubuntu-precise-keyring.gpg
```

Comment:
This is susceptible to a classic /tmp symlink attack if your kernel doesn't implement `protected_symlinks` by default (the distro kernels in distributions like Debian, Ubuntu and Fedora do this by default, but custom kernels don't necessarily, even on those distributions).